tests/builder/app_settings.rs

```diff
@@ -744,6 +744,18 @@ fn args_negate_subcommands_two_levels() {
     );
 }
 
+#[test]
+fn args_conflict_subcommands_error() {
+    let res = Command::new("todo")
+        .args_conflicts_with_subcommands(true)
+        .arg(arg!(--flag))
+        .subcommand(Command::new("sub1"))
+        .try_get_matches_from(vec!["", "--flag", "sub1"]);
+
+    assert!(res.is_err());
+    assert_eq!(res.unwrap_err().kind(), ErrorKind::ArgumentConflict);
+}
```

Comment:
I'd like to see a test that shows the error message so we can make sure its providing a desirable user experience.

`utils::assert_eq` will provide a good message for this